scilpy/image/volume_operations.py

```diff
@@ -603,6 +603,81 @@ def resample_volume(img, ref_img=None, volume_shape=None, iso_min=False,
     return nib.Nifti1Image(data2.astype(data.dtype), affine2)
 
 
+def reshape_volume(
+    img, volume_shape, mode='constant', cval=0
+):
+    """ Reshape a volume to a specified shape by padding or cropping. The
+    new volume is centered wrt the old volume in world space.
+
+    Parameters
+    ----------
+    img : nib.Nifti1Image
+        The input image.
+    volume_shape : tuple of 3 ints
+        The desired shape of the volume.
+    mode : str, optional
+        Padding mode. See np.pad for more information. Default is 'constant'.
+    cval: float, optional
+        Value to use for padding when mode is 'constant'. Default is 0.
+
+    Returns
+    -------
+    reshaped_img : nib.Nifti1Image
+        The reshaped image.
+    """
+
+    data = img.get_fdata(dtype=np.float32)
+    affine = img.affine
+
+    # Compute the difference between the desired shape and the current shape
+    diff = (np.array(volume_shape) - np.array(data.shape[:3])) // 2
```

Comment:
How hard would it be to have an option to be centered on the data's barycenter?
Would it be useful to allow to pad or crop even if the brain is not exactly center?

Reply:
I think it could be a nice option but kind of out of scope for this PR. One way to do this without any modification would be to use scil_volume_crop, then pad to the desired shape.